File: frontend/types/audit.ts
export interface AuditEvent {
  id: string;
  patientId: string | null;
  eventType: string;
  actor: string;
  description: string;
  metadata: Record<string, unknown> | null;
  createdAt: string;
}

export interface AuditListResponse {
  events: AuditEvent[];
  totalCount: number;
}

File: backend/app/schemas/audit.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class AuditEventResponse(BaseModel):
    id: str
    patientId: Optional[str]
    eventType: str
    actor: str
    description: str
    metadata: Optional[dict]
    createdAt: datetime

    model_config = {"from_attributes": True}

class AuditListResponse(BaseModel):
    events: List[AuditEventResponse]
    totalCount: int
